content/spin/serverless-ai-tutorial.md

```diff
@@ -0,0 +1,749 @@
+}
+
+const prompt = `\
+You are a bot that generates sentiment analysis responses. Respond with a single positive, negative, or neutral.
+
+Hi, my name is Bob
+neutral
+
+I am so happy today
+positive
+
+I am so sad today
+negative
+
+<SENTENCE>
+`
+
+func init() {
+	spinhttp.Handle(func(w http.ResponseWriter, r *http.Request) {
+		router := spinhttp.NewRouter()
+		router.POST("/api/sentiment-analysis", performSentimentAnalysis)
+		router.ServeHTTP(w, r)
+	})
+}
+
+func performSentimentAnalysis(w http.ResponseWriter, r *http.Request, ps spinhttp.Params) {
+	var req sentimentAnalysisRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	fmt.Printf("Performing sentiment analysis on: %q\n", req.Sentence)
+
+	// Open the KV store
+	store, err := key_value.Open("default")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer key_value.Close(store)
+
+	// If the sentiment of the sentence is already in the KV store, return it
+	exists, err := key_value.Exists(store, req.Sentence)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if exists {
+		fmt.Println("Found sentence in KV store returning cached sentiment")
+		value, err := key_value.Get(store, req.Sentence)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		res := sentimentAnalysisResponse{
+			Sentiment: string(value),
+		}
+		if err := json.NewEncoder(w).Encode(res); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		return
+	}
+	fmt.Println("Sentence not found in KV store")
+
+	// Otherwise, perform sentiment analysis
+	fmt.Println("Running inference")
+	params := &llm.InferencingParams{
+		MaxTokens:   10,
+		Temperature: 0.5,
+	}
+
+	result, err := llm.Infer("llama2-chat", strings.Replace(prompt, "<SENTENCE>", req.Sentence, 1), params)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	fmt.Printf("Inference result (%d tokens): %s\n", result.Usage.GeneratedTokenCount, result.Text)
+
+	var sentiment string
+	if fields := strings.Fields(result.Text); len(fields) > 0 {
+		sentiment = fields[0]
+	}
+
+	// Cache the result in the KV store
+	fmt.Println("Caching sentiment in KV store")
+	key_value.Set(store, req.Sentence, []byte(sentiment))
+
+	res := &sentimentAnalysisResponse{
+		Sentiment: sentiment,
+	}
+	if err := json.NewEncoder(w).Encode(res); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func main() {}
+
+```
+
+{{ blockEnd }}
+
+{{ blockEnd }}
+
+### Additional Functionality
+
+This application also includes two more components, a key/value explorer and static-fileserver component. Let's quickly go ahead and create those (letting Spin do all of the scaffolding for us).
+
+### Static Fileserver Component For The UI
```

Comment:
If you don't have a front-ned to add to this tutorial, I think it's ok skipping and just ask the user to do `curl` instead.